server/command.go

```diff
@@ -88,7 +88,7 @@ func (p *Plugin) executeCommandPreview(teamName string, args *model.CommandArgs)
 	found := false
 	for _, message := range p.getWelcomeMessages() {
 		if message.TeamName == teamName {
-			if err := p.previewWelcomeMessage(teamName, args, *message); err != nil {
+			if err := p.previewWelcomeMessage(args, *message); err != nil {
```

Comment:
While I think the behavior change is reasonable, this makes `teamName` not longer needed. Would you please remove it, also update the slash command auto complete and `README.md`?

Thanks!

Reply:
Sure, Will do.

Reply:
@hanzei I am pushing a fix for #65 which makes the above code valid. I have updated the `README` on that to include the changes.

Reply:
@hanzei I am pushing a fix for #65 which makes the above code valid. I have updated the `README` on that to include the changes.

Reply:
Would you mind including the docs and autocomplete changes in this PR? That makes it easier to review if a PR is complete-

Reply:
cc @maisnamrajusingh on the above note from hanzei

Reply:
got it guys.

Reply:
@hanzei  I think, there  is a small confusion here. There is no need to change the docs in this case as we are still using the `teamName` to compare the config values that we get from the plugin config. Please check line no 90 on the same file.

Reply:
Could you please elaborate why `teamName` is check, if `previewWelcomeMessage` uses `args.TeamId` anyway?

Reply:
@hanzei  the way the command works is `/welcomebot preview <TeamName>` so we need the team name to check which messages in the config relate to the particular team in the command.

Reply:
The team name passed to the command is not longer coupled with the `args.TeamId` used to post the preview. Hence a user could use `/welcomebot preview SomeOtherTeamName` and still get a preview for the current team.

Reply:
@hanzei  sorry I thought this pr was merged, let me look into this today and get back to you

Reply:
@maisnamrajusingh Any updates on this PR, or an ETA when it's ready for a review?

Reply:
@hanzei  sorry, it took so long to respond to this. If you look at the code, you can see that the teamName is still used to make the comparison to check if the message belongs to that team. Or did I misunderstand your comment?

```
func (p *Plugin) executeCommandPreview(teamName string, args *model.CommandArgs) {
	found := false
	for _, message := range p.getWelcomeMessages() {
		if message.TeamName == teamName {
			if err := p.previewWelcomeMessage(args, *message); err != nil {
				p.postCommandResponse(args, "error occurred while processing greeting for team `%s`: `%s`", teamName, err)
				return
			}

			found = true
		}
	}

	if !found {
		p.postCommandResponse(args, "team `%s` has not been found", teamName)
	}
}
```

Reply:
cc @hanzei quick reminder on above

Reply:
The current behavior on `master` is: The user has to pass a team name, which is used to post the message.

With your PR the behavior is: The user has to pass a team name, which is used to find the welcome message, but the template itself is executed within the context of the current team.

Either the behavior should stay the same as in `master` or the team name should not longer be required and the current team is used.

Does that make things more clear?

Reply:
@hanzei makes sense. will update the logic and update accordingly.